x-pack/plugin/ml/qa/multi-cluster-tests-with-security/src/test/resources/rest-api-spec/test/multi_cluster/50_sparse_vector.yml

```diff
@@ -184,3 +198,346 @@ teardown:
 
   - match: { hits.total.value: 5 }
   - match: { hits.hits.0._source.source_text: "the octopus comforter smells" }
+
+---
+"Check sparse_vector token pruning index_options mappings":
+
+  - requires:
+      cluster_features: 'sparse_vector_index_options_supported'
+      reason: "sparse_vector token pruning index options added support in 9.1"
+  - skip:
+      features: headers
+
+  - do:
+      headers:
+        Authorization: "Basic dGVzdF91c2VyOngtcGFjay10ZXN0LXBhc3N3b3Jk" #test_user credentials
+        Content-Type: application/json
+      indices.create:
+        index: sparse_vector_pruning_test
+        body:
+          mappings:
+            properties:
+              text:
+                type: text
+              ml.tokens:
+                type: sparse_vector
+                index_options:
+                  prune: true
+                  pruning_config:
+                    tokens_freq_ratio_threshold: 1.0
+                    tokens_weight_threshold: 0.4
+              embeddings:
+                type: sparse_vector
+
+  - match: { acknowledged: true }
+
+  - do:
+      headers:
+        Authorization: "Basic dGVzdF91c2VyOngtcGFjay10ZXN0LXBhc3N3b3Jk" #test_user credentials
+        Content-Type: application/json
+      indices.get_mapping:
+        index: sparse_vector_pruning_test
+
+  - match: { sparse_vector_pruning_test.mappings.properties.ml.properties.tokens.index_options.prune: true }
+  - match: { sparse_vector_pruning_test.mappings.properties.ml.properties.tokens.index_options.pruning_config.tokens_freq_ratio_threshold: 1.0 }
+  - match: { sparse_vector_pruning_test.mappings.properties.ml.properties.tokens.index_options.pruning_config.tokens_weight_threshold: 0.4 }
+
+---
+"Check sparse_vector token pruning index_options mappings defaults":
+
+  - requires:
+      cluster_features: 'sparse_vector_index_options_supported'
+      reason: "sparse_vector token pruning index options added support in 9.1"
+  - skip:
+      features: headers
+
+  - do:
+      headers:
+        Authorization: "Basic dGVzdF91c2VyOngtcGFjay10ZXN0LXBhc3N3b3Jk" #test_user credentials
+        Content-Type: application/json
+      indices.create:
+        index: sparse_vector_pruning_test
+        body:
+          mappings:
+            properties:
+              text:
+                type: text
+              ml.tokens:
+                type: sparse_vector
+              embeddings:
+                type: sparse_vector
+
+  - match: { acknowledged: true }
+
+  - do:
+      headers:
+        Authorization: "Basic dGVzdF91c2VyOngtcGFjay10ZXN0LXBhc3N3b3Jk" #test_user credentials
+        Content-Type: application/json
+      indices.get_mapping:
+        index: sparse_vector_pruning_test
+
+  - match: { sparse_vector_pruning_test.mappings.properties.ml.properties.tokens.index_options.prune: null }
```

Comment:
Can we update the match checks in this test to verify that index_options does not exist in the returned mappings when they shouldn't be set? Instead of `match:` this would be something like `- not_exists: sparse_vector_pruning_test.mappings.properties.ml.properties.tokens.index_options`